app/views/assignments/show.html.erb

```diff
@@ -73,12 +73,19 @@
       </div>
     <% else %>
       <% if @assignment_repos.present? %>
-        <div class="assignment-repo-list">
-          <% @assignment_repos.each do |assignment_repo| %>
-            <%= render partial: 'orgs/assignment_repos/assignment_repo', locals: { url: organization_assignment_assignment_repo_path(@organization, @assignment, assignment_repo) } %>
-          <% end %>
-        </div>
+        <%= 
+          render partial: 'shared/select_menu',
+            locals: { 
+              title: 'Sort by:',
+              options: @all_sort_modes,
+              option_links: @all_sort_modes.map { |mode| "?sort_assignment_repos_by=#{mode}" },
```

Comment:
We don't really need to pass this in. We can remove `option_links` from the locals hash, then in `shared/select_menu` line 22, do:

```ruby
link_to "?sort_assignment_repos_by=#{options[i]}"
```

It'd be even nicer if we could generate that link in a view model, but I may be being too picky 😛

Reply:
Having select-menu as a shared partial, I was thinking of re-using this like `Save as: ` button with exporting feature for multiple formats like CSV, XLS, JSON, etc. as requested in #1184. Hence, I provided a way to generally redirect to other links. Let me know if this has to be changed. 😄

Reply:
Ah right on. Makes sense to me :+1: